src/main/java/StringCalculator.java

```diff
@@ -0,0 +1,41 @@
+public class StringCalculator {
+    public static int sum(String input){
+        int output = 0;
+        int[] nums = stringToIntArray(input);
+        for(int num : nums){
+            output += num;
+        }
+        return output;
+    }
+
+    public static int[] stringToIntArray(String input){
+        String customDelimiter = "";
+        if(input.substring(0,2).equals("//")){
+            customDelimiter = input.substring(2).split("\n")[0];
+            input = input.split("\n")[1];
+        }
+        String regex = ",|:";
+        if(customDelimiter.length() > 0) {
+            if(customDelimiter.equals(".")){
+                regex += "|" + "\\.";
+            }else{
+                regex += "|" + customDelimiter;
+            }
+        }
+        String[] tokens = input.split(regex);
+        int[] res = new int[tokens.length];
+        int idx = 0;
+        for(String token : tokens){
+            try{
+                int num = Integer.parseInt(token);
+                if(num>9 || num<0){
+                    throw new RuntimeException();
+                }
+                res[idx++] = num;
+            }catch(Exception e){
+                throw new RuntimeException();
+            }
+        }
+        return res;
```

Comment:
for-each 형식을 사용해서 루프를 도신다면 stream 을 사용해서 아래와 같이 표현해볼 수 있을 것 같아요!! 😁

```suggestion
return Arrays.stream(input.split(regex))
                .mapToInt(token -> parseToValidInteger(token))
                .toArray();
    }

private static int parseToValidInteger(final String token) {
    try{
        final int num = Integer.parseInt(token);
        if(num > 9 || num < 0){
            throw new RuntimeException();
        }
        return num;
    } catch (NumberFormatException e){
        throw new RuntimeException();
    }
}
```

Reply:
stream 활용이 익숙치 않았는데, 더 좋은 방법인 것 같아요! 관련해서 추가학습 해야겠습니다ㅎㅎ